Common Tools/DeviceRecognizer.swift

```diff
@@ -0,0 +1,181 @@
+//
+//  DeviceRecognizer.swift
+//  HSRPizzaHelper
+//
+//  Created by Bill Haku on 2023/5/8.
+//  机型识别工具
+
+import Foundation
+import SwiftUI
+import UIKit
+
+// MARK: - Device
+
+enum Device {
+    // MARK: 当前设备类型 iPhone iPad Mac
+
+    enum Devicetype {
+        case iphone, ipad, mac
+    }
+
+    static var deviceType: Devicetype {
+        if UIDevice.current.userInterfaceIdiom == .pad {
+            return .ipad
+        } else {
+            return .iphone
+        }
+    }
+}
+
+// MARK: - DeviceRotationViewModifier
+
+// Our custom view modifier to track rotation and
+// call our action
+struct DeviceRotationViewModifier: ViewModifier {
+    let action: (UIDeviceOrientation) -> ()
+
+    func body(content: Content) -> some View {
+        content
+            .onAppear()
+            .onReceive(
+                NotificationCenter.default
+                    .publisher(
+                        for: UIDevice
+                            .orientationDidChangeNotification
+                    )
+            ) { _ in
+                action(UIDevice.current.orientation)
+            }
+    }
+}
+
+// A View wrapper to make the modifier easier to use
+extension View {
+    func onRotate(perform action: @escaping (UIDeviceOrientation) -> ())
+        -> some View {
+        modifier(DeviceRotationViewModifier(action: action))
+    }
+}
+
+extension UIDevice {
+    public static let modelIdentifier: String = {
+        var systemInfo = utsname()
+        uname(&systemInfo)
+        let machineMirror = Mirror(reflecting: systemInfo.machine)
+        let identifier = machineMirror.children
+            .reduce("") { identifier, element in
+                guard let value = element.value as? Int8,
+                      value != 0 else { return identifier }
+                return identifier + String(UnicodeScalar(UInt8(value)))
+            }
+        return identifier
+    }()
+
+    public static let modelName: String = {
+        var systemInfo = utsname()
+        uname(&systemInfo)
+        let machineMirror = Mirror(reflecting: systemInfo.machine)
+        let identifier = machineMirror.children
+            .reduce("") { identifier, element in
+                guard let value = element.value as? Int8,
+                      value != 0 else { return identifier }
+                return identifier + String(UnicodeScalar(UInt8(value)))
+            }
+
+        // swiftlint:disable:next cyclomatic_complexity
+        func mapToDevice(identifier: String)
```

Comment:
Superfluous Disable Command Violation: SwiftLint rule 'cyclomatic_complexity' did not trigger a violation in the disabled region; remove the disable command (superfluous_disable_command)